### PR TITLE
Add support for images in follow-hint.

### PR DESCRIPTION
**Pull request by aartaka**
This make images openable with `follow-hint`.

# What's New
- `image-hint` class that behaves mostly like `link-hint`.
- `elements-from-json` relies on `str:string-case`. It does look clearer, doesn't it? 
- `(cdr (assoc ...) ...)` in `elements-from-json` uses `alex:assoc-value` now.

# Caveats
- Naming may be inconsistent. Examples: `img` tag vs. `image-hint` class, `src` property vs. `url` slot. Is that okay?
- `image-hint` inherits from `link-hint`. This seems to be okay as long as `body` is not moved to `link-hint`. 
- By the way, why `body` slot is in `hint` not in `link-hint`?

# How Has This Been Tested
- Nyxt compiles without warnings.
- Images on protonmail.com become openable. Should work for all the other images on the Internet that are properly coded.

What do you think of it?

EDIT: `alex:assoc-value`.

### CODE REVIEW

File: source/element-hint.lisp [hunk 1]
```diff
@@ -199,29 +201,37 @@ identifier for every hinted element."
...
-                                        :body (plump:text (plump:parse (cdr (assoc :body element))))))
-                        ((equal "input" object-type)
-                         (make-instance 'input-hint
-                                        :identifier (cdr (assoc :identifier element))
-                                        :hint (cdr (assoc :hint element))))
-                        ((equal "textarea" object-type)
-                         (make-instance 'textarea-hint
-                                        :identifier (cdr (assoc :identifier element))
-                                        :hint (cdr (assoc :hint element))))))))
+                  (str:string-case object-type
```
Comment (Ambrevar) [inline source/element-hint.lisp:204]: Note that `match` would have worked here too ;)

File: source/element-hint.lisp [hunk 2]
```diff
@@ -199,29 +201,37 @@ identifier for every hinted element."
...
+                    ("textarea"
+                     (make-instance 'textarea-hint
+                                    :identifier (cdr (assoc :identifier element))
+                                    :hint (cdr (assoc :hint element))))
+                    ("img"
+                     (make-instance 'image-hint
+                                    :identifier (cdr (assoc :identifier element))
+                                    :url (cdr (assoc :src element))
+                                    :alt (cdr (assoc :alt element))
+                                    :hint (cdr (assoc :hint element))))))))
```
Comment (Ambrevar) [inline source/element-hint.lisp:229]: While you are at it, you can replace all the `(cdr (assoc ...))` with `(alex:assoc-value ...)` :)

**Comment by Ambrevar** [comment]
Thank you so much for this, I've been wanting this for so long! :)

> - Naming may be inconsistent. Examples: `img` tag vs. `image-hint` class, `src` property vs. `url` slot. Is that okay?

I prefer to keep Common Lisp code free from the harder-to-read HTML
nomenclature.
Maybe `url` could be renamed to `source` and `image-hint` to `image`?

> - `image-hint` inherits from `link-hint`. This seems to be okay as long as `body` is not moved to `link-hint`. 
> - By the way, why `body` slot is in `hint` not in `link-hint`?

I don't know.  @jmercouris?

File: source/element-hint.lisp (hunk 1 above)
Comment (aartaka) [reply source/element-hint.lisp:204]: I'm not fan of `match` -- it doesn't looks enough CL-ish to me. You're right, though -- it's a worthy alternative!

In this particular case, `str:string-case` both conveys the type information and reminds of standard `case`, so it's more informative :)

**Comment by aartaka** [comment]
OK, there're more applicatios for `body`, actually -- `button-hint` uses it too.

File: source/element-hint.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/element-hint.lisp:204]: What not CL-ish enough?  It's all s-exps, isn't it?

**Comment by Ambrevar** [comment]
So no more TODO: then?  If so, remove (or clarify) the TODO: comment you've added and we can merge :)

**Comment by aartaka** [comment]
> Maybe `url` could be renamed to `source` 

This I can agree with, but then inheriting from `link-hint` won't be possible. `url` is informative enough, so let's keep it there.

> and `image-hint` to `image`?

This I can't agree with -- `image-hint` is consistent with other `*-hint` classes, and it should be -- after all, it's not actual image we're handling, but a hint pointing at it :upside_down_face:

**Comment by aartaka** [comment]
> So no more TODO: then? If so, remove (or clarify) the TODO: comment you've added and we can merge :)

Done! :)

File: source/element-hint.lisp (hunk 1 above)
Comment (aartaka) [reply source/element-hint.lisp:204]: It indeed is. It's just that `match` have originated in Scheme and it looks foreign in CL to me. 

But these are just opinions -- John prefers the grotesque and mighty `loop` macro over any other iteration construct, for example :)

**Comment by Ambrevar** [comment]
Thanks!

**Pull request merged**